repos/iiif-storage/src/Controller/PresleyController.php

```diff
@@ -0,0 +1,168 @@
+use Zend\Diactoros\Response\JsonResponse;
+
+class PresleyController extends AbstractPsr7ActionController
+{
+
+    /**
+     * @var ManifestBuilder
+     */
+    private $builder;
+    /**
+     * @var ManifestRepository
+     */
+    private $repo;
+    /**
+     * @var CanvasRepository
+     */
+    private $canvasRepo;
+    /**
+     * @var CollectionRepository
+     */
+    private $collectionRepo;
+    /**
+     * @var Acl
+     */
+    private $acl;
+
+    public function __construct(
+        ManifestBuilder $builder,
+        ManifestRepository $repo,
+        CanvasRepository $canvasRepo,
+        CollectionRepository $collectionRepo,
+        Acl $acl
+    ) {
+        $this->builder = $builder;
+        $this->repo = $repo;
+        $this->canvasRepo = $canvasRepo;
+        $this->collectionRepo = $collectionRepo;
+        $this->acl = $acl;
+    }
+
+    /**
+     * Returns root collection with all manifest collections
+     */
+    public function getRootCollectionAction()
+    {
+        $this->allowCors();
+        $manifests = $this->repo->getNonReplacements();
+
+        return new JsonResponse(
+            [
+                "@context" => "http://iiif.io/api/presentation/2/context.json",
+                "@id" => $this->url()->fromRoute(null, [], ['force_canonical' => true], true),
+                "@type" => "sc:Collection",
+                "members" => array_map(function ($manifest) {
+                    $json = $this->builder->build($manifest, false, 0, 1)->getJson();
+                    $json['service'] = [
+                        "highlight" => "info",
+                        "profile" => "row-profile-here",
+                        "values" => [
+                            "",
+                            $json['label'],
+                            "",
+                            "",
+                            "",
+                            ""
+                        ]
+                    ];
+                    return $json;
+                }, $manifests),
+            ]
+            , 200);
+    }
+
+    /**
+     * Gets single derived manifest collection by ID
+     */
+    public function getManifestsCollectionAction()
+    {
+        $this->allowCors();
+
+        $collectionId = $this->params()->fromRoute('collection');
+
+        $manifestItem = $this->repo->getById($collectionId);
+        $manifest = $this->builder->build($manifestItem);
+
+        $manifests = $this->repo->getReplacements($collectionId);
+
+        $json = $manifest->getJson();
+        // Invalid on a collection
+        unset($json['structures']);
+        unset($json['sequences']);
+        $json['@type'] = 'sc:Collection';
+        $json['@id'] = $this->url()->fromRoute(null, [], ['force_canonical' => true], true);
+        $json['members'] = array_map(function($manifest) {
+            return $this->builder->build($manifest, false, 0, 1)->getJson();
+        }, $manifests);
+
+        return new JsonResponse($json);
+    }
```

Comment:
So the sorting room aims to split a big manifest, into smaller manifests. Sort of like:
```
- 300 page manifest
    - manifest 0-100
    - manifest 100-200
    - manifest 200-300
```

Since this doesn't really model in IIIF, the large manifest is _served_ as if it were a collection.

```
- 300 page manifest
    - "300 page manifest" collection [virtual]
        - manifest 0-100
        - manifest 100-200
        - manifest 200-300
```

The inner manifests all have a property `"dcterms:replaces": "https://example.org/manifest1"` pointing to the parent manifest. After creating the empty collection based on the manifest properties, it queries for manifests with `dcterms:replaces` that match itself. Put it all together and you have a collection.

You can do this for any manifest by going to:
```
/presley/collection/{manifest-id}
```